src/main/java/org/jenkinsci/plugins/workflow/multibranch/JobPropertyStep.java

```diff
@@ -96,39 +111,37 @@
             }
             BulkChange bc = new BulkChange(job);
             try {
-                if (!isMultibranch) {
-                    l.getLogger().println(Messages.JobPropertyStep__could_remove_warning());
-                }
                 for (JobProperty prop : job.getAllProperties()) {
                     if (prop instanceof BranchJobProperty) {
                         // TODO do we need to define an API for other properties which should not be removed?
```

Comment:
Not any more, so delete this comment. Ideally we could also delete the special-case handling of `BranchJobProperty`, but we probably need to keep it to handle some cases of upgraded branch projects: e.g., no remaining build history, or the first build after the upgrade is the one introducing `properties`. Unless you are confident that the logic below will never remove a property which was added via step.